modules/fedimint-ln-client/src/lib.rs

```diff
@@ -292,6 +302,18 @@ impl LightningClientExt for Client {
             (PayType::Lightning(operation_id), output, contract_id)
         };
 
+        // Verify that no other outgoing contract exists
+        if lightning
+            .module_api
+            .fetch_contract2(contract_id)
+            .await
+            .is_ok()
+        {
+            return Err(anyhow::anyhow!(
+                "Previous payment attempt still in progress"
+            ));
+        }
```

Comment:
I don't think it's enough to check if the contract is empty/non-existent. If called 2 very quickly in a row this fn could start the second operation before the first has funded the contract. 

(The UI sending events multiple times very quickly is actually a common source of bugs, so better to be resilient)

Reply:
Ah I see. Yes I think that is an issue, good catch. I believe the fix should be to commit the `PaymentResult` only after the contract has been successfully funded. This should move into the state machine for `pay`. Will fix.

Reply:
Added check for previous operation id, as we discussed offline.